source/mongodb/tables/role-configuration.rst

```diff
@@ -36,82 +41,149 @@
 
    * - | ``apply_when``
        | Expression
-     - An :ref:`expression <expressions>` that evaluates to ``true`` when
-       this role applies to a user for a specific document.
+     - An :ref:`expression <expressions>` that evaluates to true when
+       this :ref:`role <roles>` applies to a user for a specific document.
+
+       For a given request on a document, App Services evaluates the
+       ``apply_when`` of each relevant role until one evaluates to true.
+       Relevant roles are any roles specified for the document's collection plus
```

Comment:
I think there is a subtle detail missing here:
- When a request comes in, App Services looks for a rule matching the namespace (ie namespace rule) before looking at roles. If there is no namespace rule corresponding to the namespace of the request, then that's when we'll use the default rule's roles. If there is no matching namespace rule, AND no default rule, I believe the behavior is to deny the request
- Once the set of roles is established, they are checked in role order https://www.mongodb.com/docs/atlas/app-services/rules/roles/#role-order. If no role applies, then the behavior is to deny the request

Reply:
Namespace ~= collection, or something else entirely?

Reply:
A namespace is composed of a database and collection name; I think the terms "namespace rule" and "collection rule" are used interchangeably internally, but I think we refer to it as "Collection rules" to customers (at least in the Rules UI).

Reply:
Cool, that's what we've been saying in the docs as well. I'll stick with that for now, though it might be worth clarifying somewhere that we mean collection within a specific database (if it's not obvious already)

Reply:
(feel free to resolve this thread)